src/doc/trpl/guessing-game.md

```diff
@@ -0,0 +1,991 @@
+the user types into the `&mut String` we pass it. But it also returns
+a value: in this case, an [`io::Result`][ioresult]. Rust has a number of
+types named `Result` in its standard library: a generic [`Result`][result],
+and then specific versions for sub-libraries, like `io::Result`.
+
+[ioresult]: ../std/io/type.Result.html
+[result]: ../std/result/enum.Result.html
+
+The purpose of these `Result` types is to encode error handling information.
+Values of the `Result` type, like any type, have methods defined on them. In
+this case, `io::Result` has an `ok()` method, which says ‘we want to assume
+this value is a successful one. If not, just throw away the error
+information’. Why throw it away? Well, for a basic program, we just want to
+print a generic error, as basically any issue means we can’t continue. The
+`ok()` method returns a value which has another method defined on it:
+`expect()`. The `expect()` method takes a value it’s called on, and if it
+isn’t a successful one, [`panic!`][panic]s with a message you passed you
+passed it. A `panic!` like this will cause our program to crash, displaying
+the message.
+
+[panic]: error-handling.html
+
+If we leave off calling these two methods, our program will compile, but
+we’ll get a warning:
+
+```bash
+$ cargo build
+   Compiling guessing_game v0.1.0 (file:///home/you/projects/guessing_game)
+src/main.rs:10:5: 10:39 warning: unused result which must be used,
+#[warn(unused_must_use)] on by default
+src/main.rs:10     io::stdin().read_line(&mut guess);
+                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+```
+
+Rust warns us that we haven’t used the `Result` value. This warning comes from
+a special annotation that `io::Result` has. Rust is trying to tell you that
+you haven’t handled a possible error. The right way to suppress the error is
+to actually write error handling. Luckily, if we just want to crash if there’s
+a problem, we can use these two little methods. If we can recover from the
+error somehow, we’d do something else, but we’ll save that for a future
+project.
+
+There’s just one line of this first example left:
+
+```rust,ignore
+    println!("You guessed: {}", input);
+}
+```
+
+This prints out the string we saved our input in. The `{}`s are a placeholder,
+and so we pass it `input` as an argument. If we had multiple `{}`s, we would
+pass multiple arguments:
+
+```rust
+let x = 5;
+let y = 10;
+
+println!("x and y: {} and {}", x, y);
+```
+
+Easy.
+
+Anyway, that’s the tour. We can run what we have with `cargo run`:
+
+```bash
+$ cargo run
+   Compiling guessing_game v0.1.0 (file:///home/you/projects/guessing_game)
+     Running `target/debug/guessing_game`
+Guess the number!
+Please input your guess.
+6
+You guessed: 6
+```
+
+All right! Our first part is done: we can get input from the keyboard,
+and then print it back out.
+
+# Generating a secret number
+
+Next, we need to generate a secret number. Rust does not include random
```

Comment:
It's hard. Text like that tends to get out of date, though I guess this phrasing does too. Hrm.